homeassistant/components/izone/update_coordinator.py

```diff
@@ -0,0 +1,186 @@
+"""Data update coordinator for iZone."""
+from collections.abc import Callable
+import logging
+from typing import Any
+
+import pizone
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import (
+    DISPATCH_CONTROLLER_DISCONNECTED,
+    DISPATCH_CONTROLLER_RECONNECTED,
+    IZONE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ControllerUpdateCoordinator:
+    """Coordinate updates for a controller. This works similarly to homeassistant.helpers.update_coordinator.DataUpdateCoordinator."""
```

Comment:
Docstrings should not be longer than 72 characters per line. The first line is a header and should fit on a single line. An optional body with multiple lines can be added after a blank line.